Update third_party/flatbuffers/BUILD.bazel flatc linkopts

dbonner opened the pull request:
This is suggested for fixing issue #36170 .

I tried building after reverting the linkopts in file `third_party/flatbuffers/BUILD.bazel` to before  https://github.com/tensorflow/tensorflow/commit/adf6e22e4af83afd55e0da3caa7e7959def1e6b6 :
i.e. adding the following under “# Public flatc compiler.”
```
    linkopts = select({
         ":freebsd": [
             "-lm",
         ],
         ":windows": [],
         "//conditions:default": [
             "-lm",
             "-ldl",
         ],
     }),
```
I get the following build error:
```
ERROR: Analysis of target '//tensorflow/tools/pip_package:build_pip_package' failed; build aborted:
@flatbuffers//:freebsd is not a valid configuration key for @flatbuffers//:flatc
```
Then if I leave out the freebsd selection, I get the following build error:
```
ERROR: Analysis of target '//tensorflow/tools/pip_package:build_pip_package' failed; build aborted:
@flatbuffers//:windows is not a valid configuration key for @flatbuffers//:flatc
```
So I was able to build by just using `linkopts = [“-lm”,”-ldl”,],` under  “# Public flatc compiler.”

dbonner committed:
Update third_party/flatbuffers/BUILD.bazel flatc linkopts

This is suggested for fixing issue #36170 .

I tried building after reverting the linkopts in file `third_party/flatbuffers/BUILD.bazel` to before  https://github.com/tensorflow/tensorflow/commit/adf6e22e4af83afd55e0da3caa7e7959def1e6b6 :
i.e. adding the following under “# Public flatc compiler.”
```
    linkopts = select({
         ":freebsd": [
             "-lm",
         ],
         ":windows": [],
         "//conditions:default": [
             "-lm",
             "-ldl",
         ],
     }),
```
I get the following build error:
```
ERROR: Analysis of target '//tensorflow/tools/pip_package:build_pip_package' failed; build aborted:
@flatbuffers//:freebsd is not a valid configuration key for @flatbuffers//:flatc
```
Then if I leave out the freebsd selection, I get the following build error:
```
ERROR: Analysis of target '//tensorflow/tools/pip_package:build_pip_package' failed; build aborted:
@flatbuffers//:windows is not a valid configuration key for @flatbuffers//:flatc
```
So I was able to build by just using `linkopts = [“-lm”,”-ldl”,],` under  “# Public flatc compiler.”

byronyi commented:
That is obviously because the conditions are only effective in TF workspace, but not flatbuffers workspace.

Setting ``--linkopt=-lm --linkopt=-ldl`` in [.bazelrc](https://github.com/tensorflow/tensorflow/blob/master/.bazelrc#L346) seems to fix builds on Linux. Not sure what to do on other platforms.

byronyi commented:
@dbonner Try using the following syntax:

```
linkopts = select({
    "@org_tensorflow//tensorflow:freebsd": [
        "-lm",
    ],
    "@org_tensorflow//tensorflow:windows": [],
    "@org_tensorflow//tensorflow:android": [],
    "//conditions:default": [
        "-lm",
        "-ldl",
    ],
}),
```

dbonner committed:
third_party/flatbuffers/BUILD.bazel linkopts (revised)

Thanks @byronyi,
The build is working with your suggestion to add `@org_tensorflow//tensorflow:` to the selections in linkopts.

dbonner committed:
/third_party/flatbuffers/BUILD.bazel linkopts (revised again)

Thanks @byronyi these are the changes in your comment (which must be your latest thinking).
This is building successfully.

dbonner commented:
Thanks @byronyi . I tried the suggested changes that were in the email I got from you:
```
    linkopts = select({
        "@org_tensorflow//tensorflow:freebsd": [
	    "-lm",
        ],
        "@org_tensorflow//tensorflow:windows": [],
        "@org_tensorflow//tensorflow:android": [],
        "//conditions:default": [
	    "-lm",
	    "-ldl",
        ],
    }),
```
The build works with the above changes.  I made these changes in to commit https://github.com/tensorflow/tensorflow/pull/36202/commits/ad7b3dc4aac4f291ed087b64d7ad53a412a61904
I noticed that your comment reads differently.  I guess you must have edited it.  Your comment says: `@//tensorflow:` instead of `@org_tensorflow//tensorflow:`.  I made a third commit https://github.com/tensorflow/tensorflow/pull/36202/commits/d95511f416d1a346a0c9f14ef4f48791b156b7f4 and this also builds successfully.  So the suggested fix from the latest commit at this stage is:
```
linkopts = select({
    "@//tensorflow:freebsd": [
        "-lm",
    ],
    "@//tensorflow:windows": [],
    "//conditions:default": [
        "-lm",
        "-ldl",
    ],
}),
```

byronyi commented:
This PR is obsoleted by #36264. @dbonner should probably close it.

dbonner commented:
Yes #36264 fixes this.  Thanks :)
